delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeViewsCoordinatorSuite.scala

```diff
@@ -0,0 +1,73 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing
+
+import cats.effect.concurrent.Ref
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.cache.KeyValueStore
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.CompositeViewsFixture
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef.{ActiveViewDef, DeprecatedViewDef}
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.sdk.views.ViewRef
+import ch.epfl.bluebrain.nexus.testkit.bio.BioSuite
+import monix.bio.Task
+
+import java.util.UUID
+
+class CompositeViewsCoordinatorSuite extends BioSuite with CompositeViewsFixture {
+
+  private val existingViewRef                    = ViewRef(projectRef, nxv + "my-view")
+  private val anotherView                        = ViewRef(projectRef, nxv + "another-view")
+  private def activeView(ref: ViewRef, rev: Int) = ActiveViewDef(
+    ref,
+    UUID.randomUUID(),
+    rev,
+    viewValue
+  )
+
+  private def deprecatedView(ref: ViewRef) = DeprecatedViewDef(ref)
+
+  // Apply clean up and return the affected view if any
+  private def cleanup(newView: CompositeViewDef, cachedViews: List[ActiveViewDef]) = {
+    for {
+      // Creates the cache
+      cache        <- KeyValueStore[ViewRef, ActiveViewDef]().tapEval { c =>
+                        cachedViews.traverse { v => c.put(v.ref, v) }
+                      }
+      // The destroy action
+      destroyed    <- Ref.of[Task, Option[ActiveViewDef]](None)
+      destroy       = (active: ActiveViewDef) => destroyed.set(Some(active))
+      _            <- CompositeViewsCoordinator.cleanupCurrent(cache, newView, destroy)
+      destroyedOpt <- destroyed.get
+    } yield destroyedOpt
+
+  }
```

Comment:
A couple of things
-  Is cache a good name here? My understanding is that this is the set of running views
- If destroy is called multiple times (which could happen if there is a bug) this would not be reflected by this fixture

Reply:
* It is the same name as in the implementation
* destroy like almost all operations to start/stop indexing is meant to be idempotent

Reply:
* I updated the test so that the value can only be set once
* `KeyValueStore` has now one implementation which is a in-memory cache
* This PR aims to fix an issue in production so it is better to keep the change minimal